--- a/libcpp/system.h
+++ b/libcpp/system.h
@@ -271,16 +271,16 @@
 # include <langinfo.h>
 #endif
 
-#ifndef HAVE_SETLOCALE
-# define setlocale(category, locale) (locale)
-#endif
-
 #ifdef ENABLE_NLS
 #include <libintl.h>
 #else
 /* Stubs.  */
 # undef dgettext
 # define dgettext(package, msgid) (msgid)
+#endif
+
+#ifndef HAVE_SETLOCALE
+# define setlocale(category, locale) (locale)
 #endif
 
 #ifndef _
